### DISCUSSION TITLE
Wildcard query?

### DISCUSSION THREAD

**Discussion by johnkwaters** [FirelyTeam/firely-net-sdk#2087] [Q&A]
How can I find say all the patients with an identifier that starts with ORG-SM-* ?

**Comment by mbaltus** [comment]
You cannot, since identifiers usually use the token search parameter type, and "A token type is a parameter that provides a close to exact match search on a string of characters" and "For tokens, matches are literal (e.g. not based on subsumption or other code system features).". See http://hl7.org/fhir/search.html#token for those details.
Your best bet IMHO would be searching for the namespace those identifiers come from, and then maybe do some more client-side filtering on the results.

**Reply by johnkwaters** [reply]
So my scenario is a want to kind of sweep a FHIR server and find any records that exist and have changed since a certain date - or even just any that exist. How would I do that?

**Reply by mbaltus** [reply]
Is this related to your question about searching for Patients with specific identifiers?

If not, again depending on whether the server supports a system wide search, you could do a search using the _lastUpdated parameter to get any resource updated since a certain date.

**Reply by johnkwaters** [reply]
Yes, it's related to that. We are monitoring a FHIR server and we want to detect any 'new patients' or 'changed patients'. I can try a system wide search and see if it is supported - what would a search like that look like?

**Reply by mbaltus** [reply]
No, system wide would mean searching across all resource types, which I thought you meant with a sweep. That would be too broad for what you want here.
New or changed Patients should show up if you do a search on Patients with the _lastUpdated parameter set to a value ge or gt your last query instant. Another option could be to look into FHIR Subscriptions to get notified of changes without having to poll, but not all servers have implemented those.

**Reply by johnkwaters** [reply]
Ah - so this _lastUpdated  is diifference from _since? What's the difference? And what would a query look like that searched for any patient that had changed since a given date?

**Reply by mbaltus** [reply]
Yes, _lastUpdated is a standard search parameter for all FHIR resources, shown in the summary table on the search page: http://hl7.org/fhir/search.html. _since is not an actual search parameter, but a parameter that sometimes is defined in operations, and for example for the history interaction.

**Reply by mbaltus** [reply]
Confusing, I must admit. _lastUpdated always looks at the lastUpdated field within the resource's metadata, whereas _since could include other things to determine what would match - although that would usually also involve the lastUpdated field...

**Reply by johnkwaters** [reply]
Trying to construct a search for all patients updates since a point in time:

```
var results = fc.Search<Patient>(
    new string[] { "_lastUpdated=2022-5-15" });

```

Is there some better way to generate a query using DateTimeOffsets? Not sure how to format this and can't find any examples in the unit tests.

**Reply by johnkwaters** [reply]
And interestingly, no matter what data I add, no entries are returned in the bundle!

```
var results = fc.Search<Patient>(
    new[] { "_lastUpdated=2010-01-01" });

```

I see this being sent:

GET https://fhir-st-mpi-clinical-dev.apps.okd.healthconcourse.com/Patient?_lastUpdated=2010-01-01

Do I need to include something else in the search?
